libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ir/IrSchema.kt

```diff
@@ -62,6 +66,10 @@ internal data class IrObject(
     val deprecationReason: String?,
     val embeddedFields: List<String>,
     val mapProperties: List<IrMapProperty>,
+    /**
+     * Only contains fields that have arguments used in operations.
```

Comment:
I think it's OK to skip that comment since it's the whole premise of the IR that it matches the target and not the GraphQL schema.

We could add a more general comment how these things are decided in `IrOperationBuilder` and link from here  though.